### ISSUE TITLE
374 examples, 20 failures

### ISSUE THREAD

**Issue by juhp** [RyanGlScott/text-show#23]
```
Failures:

  tests/Spec/Data/List/NonEmptySpec.hs:27: 
  1) Spec.Data.List.NonEmpty, NonEmpty Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       0 :| []

  tests/Spec/Derived/DataFamiliesSpec.hs:49: 
  2) Spec.Derived.DataFamilies, NotAllShow Int Int Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       NASShow2 0

  tests/Spec/Derived/DataFamiliesSpec.hs:54: 
  3) Spec.Derived.DataFamilies, KindDistinguished '() Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       KindDistinguishedUnit 0 0

  tests/Spec/Derived/DataFamiliesSpec.hs:58: 
  4) Spec.Derived.DataFamilies, KindDistinguished 'True Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       KindDistinguishedBool 0 0

  tests/Spec/Derived/DatatypeContextsSpec.hs:28: 
  5) Spec.Derived.DatatypeContexts, TyCon Int Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       TyCon 0 0 0

  tests/Spec/Derived/DatatypeContextsSpec.hs:31: 
  6) Spec.Derived.DatatypeContexts, TyFamily Int Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       TyFamily 0 0 0

Failures:

  tests/Spec/Data/List/NonEmptySpec.hs:27: 
  1) Spec.Data.List.NonEmpty, NonEmpty Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       0 :| []

  tests/Spec/Derived/DataFamiliesSpec.hs:49: 
  2) Spec.Derived.DataFamilies, NotAllShow Int Int Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       NASShow2 0

  tests/Spec/Derived/DataFamiliesSpec.hs:54: 
  3) Spec.Derived.DataFamilies, KindDistinguished '() Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       KindDistinguishedUnit 0 0

  tests/Spec/Derived/DataFamiliesSpec.hs:58: 
  4) Spec.Derived.DataFamilies, KindDistinguished 'True Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       KindDistinguishedBool 0 0

  tests/Spec/Derived/DatatypeContextsSpec.hs:28: 
  5) Spec.Derived.DatatypeContexts, TyCon Int Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       TyCon 0 0 0

  tests/Spec/Derived/DatatypeContextsSpec.hs:31: 
  6) Spec.Derived.DatatypeContexts, TyFamily Int Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       TyFamily 0 0 0

  tests/Spec/Derived/ExistentialQuantificationSpec.hs:28: 
  7) Spec.Derived.ExistentialQuantification, TyCon Int Int Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       TyConForalls 0 0 0 0

  tests/Spec/Derived/ExistentialQuantificationSpec.hs:31: 
  8) Spec.Derived.ExistentialQuantification, TyFamily Int Int Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       TyFamilyForalls 0 0 0 0

  tests/Spec/Derived/MagicHashSpec.hs:29: 
  9) Spec.Derived.MagicHash, TyCon# Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       TyCon# {tcA = 0, tcB = 0, tcInt# = 0#, tcFloat# = 0.0#, tcDouble# = 0.0##, tcChar# = 'V'#, tcWord# = 0##}

  tests/Spec/Derived/MagicHashSpec.hs:34: 
  10) Spec.Derived.MagicHash, TyFamily# Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       TyFamily# {tfA = 0, tfB = 0, tfInt# = 0#, tfFloat# = 0.0#, tfDouble# = 0.0##, tfChar# = 'V'#, tfWord# = 0##}

  tests/Spec/Derived/PolyKindsSpec.hs:28: 
  11) Spec.Derived.PolyKinds, TyConCompose Either Either Either Maybe Maybe Int Int, TextShow1 instance
       Falsifiable (after 1 test and 1 shrink): 
       0
       TyConCompose (Right (Left Nothing))

  tests/Spec/Derived/PolyKindsSpec.hs:32: 
  12) Spec.Derived.PolyKinds, TyConProxy Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       TyConProxy ()

  tests/Spec/Derived/PolyKindsSpec.hs:36: 
  13) Spec.Derived.PolyKinds, TyConReallyHighKinds (,,,,) Int Int Int Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       TyConReallyHighKinds (0,0,0,0,0)

  tests/Spec/Derived/PolyKindsSpec.hs:41: 
  14) Spec.Derived.PolyKinds, TyFamilyCompose Either Either Either Maybe Maybe Int Int, TextShow1 instance
       Falsifiable (after 1 test and 1 shrink): 
       0
       TyFamilyCompose (Right (Left Nothing))

  tests/Spec/Derived/PolyKindsSpec.hs:45: 
  15) Spec.Derived.PolyKinds, TyFamilyProxy Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       TyFamilyProxy ()

  tests/Spec/Derived/PolyKindsSpec.hs:49: 
  16) Spec.Derived.PolyKinds, TyFamilyReallyHighKinds (,,,,) Int Int Int Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       TyFamilyReallyHighKinds (0,0,0,0,0)

  tests/Spec/Derived/RankNTypesSpec.hs:28: 
  17) Spec.Derived.RankNTypes, TyCon Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       TyCon (Tagged2 0) (Tagged2 0)

  tests/Spec/Derived/RankNTypesSpec.hs:31: 
  18) Spec.Derived.RankNTypes, TyFamily Int Int, TextShow2 instance
       Falsifiable (after 1 test): 
       0
       TyFamily (Tagged2 0) (Tagged2 0)

  tests/Spec/Derived/RecordsSpec.hs:28: 
  19) Spec.Derived.Records, TyCon Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
      (:@:) {tc3 = 0, tc4 = 0}

  tests/Spec/Derived/RecordsSpec.hs:33: 
  20) Spec.Derived.Records, TyFamily Int Int, TextShow1 instance
       Falsifiable (after 1 test): 
       0
       (:!:) {tf3 = 0, tf4 = 0}

Randomized with seed 2005729184

Finished in 1.0413 seconds
374 examples, 20 failures
```

**Comment by RyanGlScott** [comment]
I was hoping to fix this before you noticed it :)

This is due to an issue in `deriving-compat-0.3.2` that I fixed with the release of `deriving-compat-0.3.3` (I've since deprecated the former version on Hackage). Can you confirm that the tests now work for you?

**Comment by juhp** [comment]
Thanks
